### PR TITLE
Refactor discrete curves using FiberBundle and QuotientMetric 

### PR DESCRIPTION
**Pull request by alebrigant**
This PR addresses issue #1183.

- It uses `FiberBundle` and `QuotientMetric` to structure the code: (1) it defines a new class `SRVShapeBundle` that inherits from `FiberBundle` in which the methods linked to verticality and horizontality (previously in `QuotientSRVMetric`) are moved. (2) It makes `QuotientSRVMetric` inherit from `QuotientMetric`. No changes were made to the classes `FiberBundle` and `QuotientMetric`, except some renaming. 

- It cleans the API for the discrete curves module.

### CODE REVIEW

File: geomstats/geometry/discrete_curves.py [hunk 1]
```diff
@@ -1772,66 +1743,73 @@ def space_derivative(curve):
         return space_deriv
 
 
-class QuotientSRVMetric(SRVMetric):
-    """Metric on shape space induced by the SRV metric.
+class SRVShapeBundle(DiscreteCurves, FiberBundle):
```
Comment (ninamiolane) [inline geomstats/geometry/discrete_curves.py:1746]: NIT (not necessarily for this PR): I wonder if we should put the name of the group inducing the quotient in the name of the class.

We could have a SRVShapeBundle only by quotienting the action of rotations.

Actually, in these shape manifolds, we will have a series of quotient, for each additional group applied.

Thus: `SRVRotationBundle`, `SRVParamBundle`?

--> I'll ask the question in our issue on DiscreteCurves.

File: geomstats/geometry/discrete_curves.py [hunk 2]
```diff
@@ -1694,17 +1665,17 @@ def path(t):
             t = gs.cast(t, gs.float32)
             t = gs.to_ndarray(t, to_ndim=1)
             t = gs.to_ndarray(t, to_ndim=2, axis=1)
-            new_initial_curve = gs.to_ndarray(initial_curve, to_ndim=curve_ndim + 1)
+            new_initial_point = gs.to_ndarray(initial_point, to_ndim=point_ndim + 1)
             new_initial_tangent_vec = gs.to_ndarray(
-                initial_tangent_vec, to_ndim=curve_ndim + 1
+                initial_tangent_vec, to_ndim=point_ndim + 1
```
Comment (ninamiolane) [inline geomstats/geometry/discrete_curves.py:1670]: Thank you so much for the renamings curve -> point, it was overdue!

We should also think about a strategy on how to handle these APIs:
- from the codebase perspective, we want to use `point` everywhere,
- from the user's perspective, they probably want to see `curve`
---> We could write an "API remapping" for the manifolds where this happens.

File: geomstats/geometry/discrete_curves.py [hunk 3]
```diff
@@ -1887,18 +1865,40 @@ def split_horizontal_vertical(self, tangent_vec, curve):
...
         tangent_vec_ver = gs.squeeze(tangent_vec_ver)
-        tangent_vec_hor = tangent_vec - tangent_vec_ver
+        if return_norm:
+            return tangent_vec_ver, vertical_norm
+
+        return tangent_vec_ver
 
-        return tangent_vec_hor, tangent_vec_ver, vertical_norm
+    def horizontal_projection(self, tangent_vec, point):
+        """Compute vertical part of tangent vector at base point.
```
Comment (ninamiolane) [inline geomstats/geometry/discrete_curves.py:1881]: NIT: Compute horizontal part

**Review by ninamiolane — APPROVED** [review]
LGTM, thank you very much! Just one docstring where vertical --> horizontal, and then one question about naming that we do not have to tackle now, I'll put it on our issue on the refactor of discrete curves :))

**Comment by ninamiolane** [comment]
Merging now, to avoid github conflicts since it's been 7 days, I'll fix the docstring typo.

File: geomstats/geometry/discrete_curves.py (hunk 2 above)
Comment (alebrigant) [reply geomstats/geometry/discrete_curves.py:1670]: Yes I was also thinking of a similar "API remapping" solution, I completely agree it would be helpful.

**Pull request merged**